Source/Scene/Camera.js

```diff
@@ -3230,9 +3240,35 @@ define([
         Matrix4.clone(camera._transform, result.transform);
         result._transformChanged = true;
 
+        if (defined(camera.cameraLimiter)) {
+            result.cameraLimiter = CameraLimiter.clone(camera.cameraLimiter, result.cameraLimiter);
+        }
+
         return result;
     };
 
+    var saveOrientation = new HeadingPitchRoll();
+    function limitPosition(camera) {
+        // following the convention from _adjustHeightForTerrain to maintain correct orientation
+        // before and after position is changed
+        if (defined(camera.cameraLimiter) && camera._mode === SceneMode.SCENE3D) {
```

Comment:
Instead of putting everything in an `if` block, we prefer to return early.  ie
```js
if (!defined(camera.cameraLimiter) || camera._mode !== SceneMode.SCENE3D) {
    return;
}
// continue limiting logic here

Reply:
Why does this only work in `SCENE3D`?  I would expect this to work in all 3 scene modes

Reply:
the current implementation does not support other scene modes - this is an issue that still needs to be resolved

Reply:
Gotcha.  I think that needs to be resolved before this gets merged.  If you need help figuring it out, you can talk to me or @bagnell